httpserver/server.go

```diff
@@ -192,3 +189,34 @@ type ServerAddrNotFoundError struct{}
 func (err ServerAddrNotFoundError) Error() string {
 	return "The http server address configuration is not set. Skipping server creation"
 }
+
+func serverGracefulShutdown(isTLSEnabled bool, svr *http.Server, lsnr net.Listener, certFile string, keyFile string) error {
+	connectionsClosed := make(chan struct{})
+	go func() {
+		sigint := make(chan os.Signal, 1)
+		signal.Notify(sigint, os.Interrupt, syscall.SIGTERM)
+		<-sigint
+		logrus.Info("shutting down ratify server...")
+		ctx, cancel := context.WithTimeout(context.Background(), 6*time.Second) // wait 6 seconds for existing connections to finish
```

Comment:
question here: what happens to new connections during this 6sec?

Reply:
ah Got it, so in a upgrade scenario, there will be maybe some "connection refused"  if there is request coming in? Is this worth documenting somewhere?

Reply:
Yes there still could be some connection refused errors but very rarely. In upgrade scenario, the old pod continues to run until the new pod is in ready state. Only then the sigterm is sent to the old pod. However, we still have a gap in container readiness. We don't have a readiness probe defined so by default the container is in "ready" state as soon as it's created. This means ratiy service will begin routing requests to the new ratify container before the http server finishes starting up. This scenario is typically a matter of 1-2 seconds but still it's possible requests get dropped during this time. 

The solution is to implement a readiness probe that will return 200 OK only once the server and tls watchers startup.